File: src/utils/teddyGenerator.js
import { supabase } from '../lib/supabase';
import { generateTeddyImage, saveTeddyImage } from './imageGenerator';

export const generateTeddyBear = async () => {
  const names = ["Whiskey Whiskers", "Madame Mistletoe", "Baron Von Blubber", "Icy Ivan", "Lady Lush"];
  const titles = ["The Smooth Operator", "The Festive Flirt", "The Inflated Ego", "The Frosty Fighter", "The Party Animal"];
  const specialMoves = ["On the Rocks", "Sneak Kiss", "Burst Bubble", "Ice Age", "Drunken Master"];

  const randomIndex = Math.floor(Math.random() * names.length);
  const name = names[randomIndex];
  const title = titles[randomIndex];
  const specialMove = specialMoves[randomIndex];
  const description = `A cheeky teddy with a knack for ${specialMove.toLowerCase()}.`;

  const imageUrl = await generateTeddyImage(name, description);
  const savedImageUrl = await saveTeddyImage(name, imageUrl);

  return {
    name,
    title,
    description,
    attack: Math.floor(Math.random() * 3) + 4, // 4-6
    defense: Math.floor(Math.random() * 3) + 4, // 4-6
    special_move: specialMove,
    image_url: savedImageUrl
  };
};

File: src/utils/teddyGenerator.py
import random
from .supabaseSetup import supabase

def generateTeddyBear():
    names = ["Whiskey Whiskers", "Madame Mistletoe", "Baron Von Blubber", "Icy Ivan", "Lady Lush"]
    titles = ["The Smooth Operator", "The Festive Flirt", "The Inflated Ego", "The Frosty Fighter", "The Party Animal"]
    
    name = random.choice(names)
    title = random.choice(titles)
    description = f"A cheeky teddy bear with a knack for mischief. Known as {title}."
    
    return {
        "name": name,
        "title": title,
        "description": description,
        "attack": random.randint(1, 10),
        "defense": random.randint(1, 10),
        "special_move": f"{name}'s Special Move",
        "image_url": f"https://example.com/{name.lower().replace(' ', '_')}.png"
    }

def generateInitialBears(count):
    bears = [generateTeddyBear() for _ in range(count)]
    supabase.table("terrible_teddies").insert(bears).execute()
    return bears